dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -886,24 +891,66 @@ default String getDockerToken(String registryDockerPath, String repo) throws URI
      * @param repo
      * @param specifierName Value of the specifier. Either a tag or a digest
      * @return HttpResponse
-     * @throws URISyntaxException
-     * @throws IOException
-     * @throws InterruptedException
      */
-    default HttpResponse<String> getDockerManifest(String token, String registryDockerPath, String repo, String specifierName)
-            throws URISyntaxException, IOException, InterruptedException {
+    default Optional<HttpResponse<String>> getDockerManifest(String token, String registryDockerPath, String repo, String specifierName) {
         // Ex: https://ghcr.io/v2/<repo>/manifests/<tag_or_digest>
         String getManifestURL = String.format("https://%s/v2/%s/manifests/%s", registryDockerPath, repo, specifierName);
         String acceptHeader = String.join(",", DOCKER_V2_IMAGE_MANIFEST_MEDIA_TYPE, DOCKER_V2_IMAGE_MANIFEST_LIST_MEDIA_TYPE, OCI_IMAGE_MANIFEST_MEDIA_TYPE, OCI_IMAGE_INDEX_MEDIA_TYPE);
+        int tooManyRequestsCode = 429;
+
+        HttpResponse<String> manifestResponse;
+        try {
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(new URI(getManifestURL))
+                    .header(HttpHeaders.ACCEPT, acceptHeader)
+                    .header(HttpHeaders.AUTHORIZATION, String.join(" ", JWT_SECURITY_DEFINITION_NAME, token))
+                    .GET()
+                    .build();
+
+            // Send request and retry if rate limit exceeded. This seems to happen occasionally for Amazon ECR images
+            boolean success = false;
+            int maxRetries = 3;
+            int retries = 0;
+            do {
```

Comment:
Neat, I think I had intended on implementing this at some point but never got around to it. 
https://github.com/dockstore/dockstore/blob/1.11.5/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ResourceUtilities.java#L78

Might be useful as a general utility